### DISCUSSION TITLE
How to label SER and RE labelling with own invoice dataset in FUND format?

### DISCUSSION THREAD

**Discussion by Gopipatwa** [PaddlePaddle/PaddleOCR#7652] [Q&A] [answered]
How to label SER and RE labelling with own invoice dataset in FUND format?

**Comment by littletomatodonkey** [comment]
Hi, thanks for your attention!

* The data format of SER and RE can be seen here: [link](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/doc_en/kie_en.md#12-custom-dataset), you need to add `label` for SER task, add `id` and `linking` for RE task.
* PPOCRLabel supports SER/KIE annotation, you can refer to [link](https://github.com/PaddlePaddle/PaddleOCR/tree/release/2.6/PPOCRLabel). Labeling for RE is on the way.

What's more, you can just use `OCR + SER` pipeline to finsh you key information extraction task, you can refer to [link](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/ppstructure/kie/how_to_do_kie_en.md).

**Reply by Gopipatwa** [reply]
Is there any tool for annotations for SER and RE dataset because PaddleocrLabel.py they have only this structure output
{transcripts:"text",points:[[]],difficult:false,key_cls:None}
And I want like this
" image path                 annotation information "
zh_train_0.jpg   [{"transcription": "汇丰晋信", "label": "other", "points": [[104, 114], [530, 114], [530, 175], [104, 175]], "id": 1, "linking": []}]

**Comment by littletomatodonkey** [comment, marked as answer]
PPOCRLabel supports KIE labeling: https://github.com/PaddlePaddle/PaddleOCR/blob/dygraph/PPOCRLabel/README.md

For RE, you can refer to this PR by community contributors: https://github.com/PaddlePaddle/PaddleOCR/pull/6837/files
